cmd/cluster/core/create_test.go

```diff
@@ -203,6 +205,16 @@ func TestValidate(t *testing.T) {
 			},
 			expectedErr: "",
 		},
+		{
+			name: "passes with KubeAPIServerDNSName",
```

Comment:
can we have a negative case, flag input for an invalid dns name?

Reply:
Humm we're not validating that opt in the validate function because we rely in the apimachinery to do that validation at API level.

Reply:
the field is validated with CEL at the API level. What I'm asking in this PR now is to validate the CLI input in func (opts *RawCreateOptions) Validate(ctx context.Context) to fail earlier and cover that in the unit.
just like we do for other fields see https://github.com/openshift/hypershift/blob/627d5bc233260e418706c139ef8b5c7cd781bdde/cmd/cluster/core/create.go#L620-L626